source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/Util.java

```diff
@@ -349,6 +353,20 @@ public static<T extends BaseActionElement> T tryCastTo(BaseActionElement actionE
         }
     }
 
+
+    /**
+     * Converts pixels to dps.
+     *
+     * @param context the Context for getting the resources
+     * @param px      dimension in pixels
+     * @return dimension in dps
+     */
+    public static @Dimension(unit = DP) int pixelToDp(@NonNull Context context, @Px int px) {
+        DisplayMetrics displayMetrics = context.getResources()
+            .getDisplayMetrics();
+        return Math.round(px / displayMetrics.density);
```

Comment:
just a question to be resolved by you if you think we need to have a fallback value in case we get undefined values? (IE: 0, negative, etc)

Reply:
@hggz 
<= 0 will mean something went wrong getting the container's width. We will not use responsive layout in that cause.